028-service-binding.md

```diff
@@ -0,0 +1,412 @@
+
+### Binding Secret in Status
+
+The Service Binding specification requires that a [provisioned service](https://github.com/servicebinding/spec#provisioned-service) includes in its status a field called `binding`. Where this field contains a single field `name` that refers to the name of a secret that should be mounted to any application that requests to bind to the service. The specification then further details recommended formats for that secret.
+
+```yaml
+status:
+  binding:
+    name: # secret name
+```
+
+### Secret reference in ServiceBinding resource
+
+The Service Binding specification requires that an application is bound to a service using a Custom Resource of kind `ServiceBinding`. The [ServiceBinding](https://github.com/servicebinding/spec#service-binding) includes a `workload` section that defines the application that is to be bound and a `service` section that describes the service the application is bound to.
+
+``` yaml
+apiVersion: servicebinding.io/v1alpha3
+kind: ServiceBinding
+metadata:
+  name: barista-kafka
+spec:
+  workload:
+    apiVersion: apps/v1
+    kind: Deployments
+    name: barista-kafka
+  service:
+    apiVersion: kafka.strimzi.io/v1beta2
+    kind: Kafka
+    name: my-cluster
+```
+
+If the service being bound does not contain a `status.binding` field the `ServiceBinding` can directly reference the secret to use in one of two ways:
+
+- Including a `spec.env` section that refers to a `Secret` that is made available by the service
+- Replacing the `spec.service` section with a direct reference to a `Secret` that is made available by the service
+
+#### Env Mapping Example
+
+```yaml
+apiVersion: servicebinding.io/v1alpha3
+kind: ServiceBinding
+metadata:
+  name: barista-kafka
+spec:
+  workload:
+    apiVersion: apps/v1
+    kind: Deployments
+    name: barista-kafka
+  service:
+    apiVersion: kafka.strimzi.io/v1beta2
+    kind: Kafka
+    name: my-cluster
+  env:
+    name: my-user
+    key: password
+```
+
+#### Direct Secret Reference Example
+
+```yaml
+apiVersion: servicebinding.io/v1alpha3
+kind: ServiceBinding
+metadata:
+  name: barista-kafka
+spec:
+  workload:
+    apiVersion: apps/v1
+    kind: Deployments
+    name: barista-kafka
+  service:
+    apiVersion: v1
+    kind: Secret
+    name: my-user
+```
+
+## Connecting to Strimzi
+
+To connect to Strimzi, a service binding needs the following:
+
+* **host** - hostname or IP address
+* **port** - port number
+* **userName** - username, optional
+* **password** - password or token, optional
+* **certificate** - certificate, optional
+
+Some of this comes from the `Kafka` CR and some from `KafkaUser`. Below is detailed the requirements for different configurations of Strimzi and why the current behaviour does not conform to the Service Binding spec.
+
+## Binding to a Kafka cluster with no TLS or authentication
```

Comment:
I included it initially so that those who are unfamiliar with Strimzi (primarily those from the Service Binding spec), understood the starting point of what we have at the moment. Since we selected an approach already I can remove these sections if you feel they are no longer required?

Reply:
It seems disconnected they way it is included right now - it does not seem to link to anything particular. So I would say we should do one of:
* Just delete it
* Merge this into the rejected alternatives since this explains to some extent the logic when coming up with them and the current state
* Give it some _introduction_ to explain what it is and why is it here

It is not a big deal ... I can live with it as it is as well if needed. It just seemed to stand out to me as a weird part - almost if it was something what was supposed to be deleted but wasn't.

Reply:
I've been finding it useful to refer back to to remember the current state of affairs with where all the connection details are while we discuss, so I've condensed it down and written an introduction to explain why it's there.